comp/otelcol/otlp/components/connector/datadogconnector/factory.go

```diff
@@ -19,15 +19,25 @@ import (
 )
 
 type factory struct {
-	tagger tagger.Component
+	tagger   testutil.TaggerClient
+	hostname option.Option[string]
 }
 
+// SourceProviderFunc is a function that returns the source of the host.
+type SourceProviderFunc func(context.Context) (string, error)
+
 // NewFactoryForAgent creates a factory for datadog connector for use in OTel agent
-func NewFactoryForAgent(tagger tagger.Component) connector.Factory {
+func NewFactoryForAgent(tagger testutil.TaggerClient, hostGetter SourceProviderFunc) connector.Factory {
```

Comment:
Implementation code should not depend on testutil, can you move the TaggerClient to e.g. traceutil/otel_util ?

Reply:
Actually I don't quite understand why you need to change it from tagger.Component to testutil.TaggerClient, is it a typo?